test/test_quantization.py

```diff
@@ -485,20 +485,8 @@ def test_quantized_rnn(self):
 
         ref = copy.deepcopy(cell)
 
-        qconfig_dynamic_dict = {
-            torch.nn.LSTM: default_dynamic_qconfig,
-        }
-        default_dynamic_module_mapping = {
-            torch.nn.LSTM: torch.nn.quantized.dynamic.LSTM,
-        }
-        model_int8 = quantize_dynamic(
-            model=model, qconfig_dict=qconfig_dynamic_dict, mapping=default_dynamic_module_mapping,
-            dtype=torch.qint8
-        )
-        model_fp16 = quantize_dynamic(
-            model=model, qconfig_dict=qconfig_dynamic_dict, mapping=default_dynamic_module_mapping,
-            dtype=torch.float16
-        )
+        model_int8 = quantize_dynamic(model=model, dtype=torch.qint8)
+        model_fp16 = quantize_dynamic(model=model, dtype=torch.float16)
```

Comment:
It would be great to also have test coverage for using the new quantize_dynamic API with NoopObserver to communicate that weights are quantized to float16. This will also serve as an example for developers on how to use `observer` mechanism for FP16 dynamic quantization.

Reply:
I'm actually fine to hide NoopObserver for common cases and use this `dtype` variant only. There's also float16_qconfig_dynamic which wraps the same. This diff is more about keeping the internal flow cleaner.